rmf_fleet_adapter/src/rmf_fleet_adapter/TaskManager.cpp

```diff
@@ -1628,6 +1628,10 @@ void TaskManager::retreat_to_charger()
 
   if (!task_planner->configuration().constraints().drain_battery())
     return;
+  if (!task_planner->configuration().constraints().retreat_to_charger())
+    return;
+  if (_idle_task && _idle_task->request_type() == "Charge")
```

Comment:
We shouldn't have comparison to magic stings like this. The `rmf_task::Request` class should have a virtual static member function that should return the request type. Then you can call `_idle_task && _idle_task->request_type() == ParkRobotIndefinitely::request_type()`

Reply:
With the proposed refactor, we no longer rely on checking request types.